lib/improver/tests/ensemble_calibration/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py

```diff
@@ -58,7 +60,18 @@ def test_basic(self):
         result = str(Plugin())
         msg = ("<ContinuousRankedProbabilityScoreMinimisers: "
                "minimisation_dict: {'gaussian': 'normal_crps_minimiser', "
-               "'truncated gaussian': 'truncated_normal_crps_minimiser'}>")
+               "'truncated gaussian': 'truncated_normal_crps_minimiser'}; "
+               "max_iterations: 1000; decimals: 5>")
+        self.assertEqual(result, msg)
+
+    def test_update_max_iterations_and_decimals(self):
+        """A test to update the max_iterations and decimals
+        keyword argument."""
+        result = str(Plugin(max_iterations=10, decimals=2))
+        msg = ("<ContinuousRankedProbabilityScoreMinimisers: "
+               "minimisation_dict: {'gaussian': 'normal_crps_minimiser', "
+               "'truncated gaussian': 'truncated_normal_crps_minimiser'}; "
+               "max_iterations: 10; decimals: 2>")
```

Comment:
I'd be tempted to scrap all of these methods to be honest. Unless we have a very good reason, we shouldn't be testing private/double underscore methods.

Reply:
I guess my concern with this would be that, for example, with the `__repr__` methods, there would be little point in having the methods without the tests, as I think we would be very unlikely to keep the `__repr__` methods up-to-date with unit test failures to remind us that the `__repr__` methods need updating. I've left these tests in, as it fits with our general "test everything" style.

Reply:
Summary: I disagree, but in this case happy to leave them in. 😄

Detail:
> I think we would be very unlikely to keep the __repr__ methods up-to-date with unit test failures to remind us that the __repr__ methods need updating

If we don't have any tests, then there's nothing that would fail (1.) - we're just adding more code to maintain here. I think that the string representation is arguably only really useful in a documentation setting, so more appropriate for a doctest if you're going to use anything.

> it fits with our general "test everything" style.

We shouldn't be trying to do this. (1.)

1. Double underscore methods/private methods should typically be reserved for implementation details - this means that, by creating lots of tests for these, you're tying yourself very strongly to a particular implementation, which leads to fragile test suites (which break often), and a higher maintenance cost. Generally, I think that want to focus your tests on the public interface (2.) (which by their nature, will test the implementation). i.e: an outside-in testing approach.

2. The string representation could be considered a public "interface", so I can see why you'd want to test it - but as I said, I think doctests would be much more suitable. I guess the line could be blurred here, and so I could support leaving it in (in the absence of doctests).

Reply:
@arh89 this strikes me as something to come back to in the wider context - joint team meetings / other discussion / training?